airflow/ti_deps/ti_deps.py

```diff
@@ -0,0 +1,479 @@
+from abc import ABCMeta, abstractmethod
```

Comment:
This kind of stuff is usually done by conventions in python. An abstract class is often prefixed by `Base`, and abstract method can just be method that `raise NotImplementedError()`